cpp/include/Ice/ClientAuthenticationOptions.h

```diff
@@ -0,0 +1,414 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+#ifndef ICE_SSL_CLIENT_AUTHENTICATION_OPTIONS_H
+#define ICE_SSL_CLIENT_AUTHENTICATION_OPTIONS_H
+
+#include "SSLConfig.h"
+#include "SSLConnectionInfo.h"
+
+#include <functional>
+
+namespace Ice::SSL
+{
+    /**
+     * The SSL configuration properties for client connections.
+     */
+    struct ClientAuthenticationOptions
+    {
+#if defined(_WIN32)
+        /**
+         * A callback that allows selecting the client's SSL certificate based on the target server host name.
```

Comment:
This target server host name is the host name in the endpoint you're using to establish the connection?

If yes, I think something like "based on the hostname or IP address in the endpoint used to establish the connection" would be clearer.

Reply:
Yes it is.